File: sample_example/src/main/java/org/example/di/AppComponent.java
package org.example.di;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.interfaces.IComponent;

@Component
public interface AppComponent extends IComponent {

    AppModule app();

    DataModule data();

    DomainModule domain();

    PresenterModule presenter();

}

File: test_feature_core_deps/src/main/java/com/github/klee0kai/test/app/di/AppComponent.kt
package com.github.klee0kai.test.app.di

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.stone.annotations.component.ExtendOf
import com.github.klee0kai.test.app.di.dependencies.AppDependencyProvider
import com.github.klee0kai.test.core.di.CoreComponent
import com.github.klee0kai.test.core.di.wrapper.CustomWrappersStone


@Component(
    wrapperProviders = [
        CustomWrappersStone::class,
    ],
)
// TODO https://github.com/klee0kai/stone/issues/116
//  make the use of interfaces position independent
interface AppComponent : AppComponentModules, AppDependencyProvider, CoreComponent {

    @ExtendOf
    fun ext(core: CoreComponent)

}